python-package/lightgbm/basic.py

```diff
@@ -612,7 +616,7 @@ def _c_int_array(data):
     return (ptr_data, type_data, data)  # return `data` to avoid the temporary copy is freed
 
 
-def _is_allowed_numpy_dtype(dtype) -> bool:
+def _is_allowed_numpy_dtype(dtype: type) -> bool:
```

Comment:
why is this one just `type`?

Setting it to `np.typing.DTypeLike` results in these `mypy` errors.

```text
python-package/lightgbm/basic.py:618: error: Argument 1 to "issubclass" has incompatible type "Union[dtype[Any], None, Type[Any], _SupportsDType[dtype[Any]], str, Union[Tuple[Any, int], Tuple[Any, Union[SupportsIndex, Sequence[SupportsIndex]]], List[Any], _DTypeDict, Tuple[Any, Any]]]"; expected "type"  [arg-type]
python-package/lightgbm/basic.py:619: error: Argument 1 to "issubclass" has incompatible type "Union[dtype[Any], None, Type[Any], _SupportsDType[dtype[Any]], str, Union[Tuple[Any, int], Tuple[Any, Union[SupportsIndex, Sequence[SupportsIndex]]], List[Any], _DTypeDict, Tuple[Any, Any]]]"; expected "type"  [arg-type]
```

And that's because it's not being passed a `dtype`... it's being passed a `type`.

https://github.com/microsoft/LightGBM/blob/1585ee1566631fca55cd3a391074f807b21bf9b5/python-package/lightgbm/basic.py#L280

https://github.com/microsoft/LightGBM/blob/1585ee1566631fca55cd3a391074f807b21bf9b5/python-package/lightgbm/basic.py#L624-L628

```python
import pandas as pd
s = pd.Series([1, 2, 3, 4, 5], dtype="Int64")
for column_name, pandas_dtype in s.to_frame().dtypes.items():
    print(pandas_dtype)
    print(type(pandas_dtype))

# Int64
# <class 'pandas.core.arrays.integer.Int64Dtype'>
```